lib/configurators/aws/ec2discovery.go

```diff
@@ -14,10 +14,25 @@
 
 package aws
 
-import "fmt"
+import (
+	crand "crypto/rand"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	mrand "math/rand"
+	"time"
+)
 
 func EC2DiscoverySSMDocument(proxy string) string {
-	return fmt.Sprintf(ec2DiscoverySSMDocument, proxy)
+	randomBytes := make([]byte, 4)
+	if _, err := crand.Read(randomBytes); err != nil {
+		// on error from crypto rand fallback to less secure math random
```

Comment:
Is this a good idea? Should we change the return to include an error, or maybe panic instead?

Alternatively we could push the "problem" to uuid.New, like Tim suggested below.

Reply:
I only see one "chain" of production callers for this method, so changing the signature ain't that bad.

Reply:
I switched to just using `uuid.New()`